protocol/avdl/keybase1/prove_common.avdl

```diff
@@ -93,6 +93,23 @@ protocol proveCommon {
     string id;
   }
 
+  record SelectorEntry {
+    // Exactly one of Is* is true
+    boolean isIndex;
+    int index;
+    boolean isKey;
+    string key;
+    boolean isAll;
+    boolean isContents;
+  }
```

Comment:
I don't know about this move. isAll and isContents don't apply to parameterized proofs, so parameterized proofs need only union{string int}. SelectorEntry is a pretty broad name to take in keybase1.

Reply:
name change would be good. i think to simplify the handling code it's best to leave this as is even if there are unused fields.

Reply:
move the json_helpers out of libkb to a separate package. unfortunately we can't move the `SelectorEntry` out of keybase1 since we need it to parse the config entry.